src/hb_persistent.erl

```diff
@@ -210,6 +210,8 @@ notify(GroupName, Msg2, Msg3, Opts) ->
         false ->
             ok
     end,
+    % Call notification dispatch for external listeners (non-blocking)
+    dispatch_to_notify_device(GroupName, Msg2, Msg3, Opts),
```

Comment:
WDYT about implementing a `dev_hook:on` call here for an `on-notify` event, then moving `dispatch_to_notify_device` to `~notify@1.0/dispatch` (or similar)? That way it is generic (which seems to be where you are going with the `notify_device` idea in `hb_opts`), and we could just add the handler in `dev_notify` as a default in the opts?